### ISSUE TITLE
Propagate Job execution Exception gracefully out of scheduler

### ISSUE THREAD

**Issue by cdaringe** [quartz-scheduler/quartz#177]
# problem statement

- an unrecoverable error has occurred in `Job::execute`, but quartz provides no means to crash process -- it swallows Exceptions and continues on happily

# discussion

- quartz should give me option to abort and shut it all down
- from the website:

> Throwing Exceptions

> A Job’s execute method should contain a try-catch block that handles all possible exceptions.

> If a job throws an exception, Quartz will typically immediately re-execute it (and it will likely throw  the same exception again). It’s better if the job catches all exception it may encounter, handle them, and reschedule itself, or other jobs. to work around the issue.

this is infeasible in my circumstances.  i have a set of knowns, and if those conditions are violated, i want the Exception to propagate up and my exception to enter my log/crash sequence.

Thoughts?

Thanks.

**Comment by bojingo** [comment]
Yes, this would be great. We have a process that runs continuously on a server as a background task and invokes Quartz jobs based on user-specified frequency configurations. We recently had reports from users that their jobs aren't running (the jobs send configurable emails, and they weren't getting their emails).

The root cause of the problem is that we updated a dependency of the email component which was causing the send to fail. This is a catastrophic issue that would prevent the background process from *ever* succeeding. I need a way to detect this sort of unrecoverable failure and just end everything until we can resolve the problem.

We have monitoring in place to ensure the background process is running "successfully" over time and the fact that Quarts suppresses these unrecoverable exceptions means we don't have as clear of a mechanism as we'd like to know something is terribly wrong. The mechanism of simply monitoring the running processes works great for our many other background processes that don't rely on Quarts. Instead we may be forced to have to build out additional log monitoring, etc. around just this background process handling our Quartz jobs.

Bottom line: I should have more control over what to do with exceptions resulting from the Quartz jobs in **my** process.

**Comment by ricardogaspar2** [comment]
Yes this feature would be great so we can have more control over the jobs and decide if the main program should or not be stopped.
